### PR TITLE
Update Laravel Scout and its dependencies

### PR DESCRIPTION
**Pull request by burakcakirel**
refs #92

### CODE REVIEW

File: src/Jobs/ImportStages.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ public static function fromSource(ImportSource $source)
     {
         $index = Index::fromSource($source);
 
-        return (new static([
+        return (new self([
```
Comment (burakcakirel) [inline src/Jobs/ImportStages.php]: https://github.com/timber/timber/pull/2203#issuecomment-591332533

File: composer.json [hunk 2]
```diff
@@ -17,17 +17,17 @@
...
+        "laravel/scout": "^8.0",
         "ongr/elasticsearch-dsl": "^7.0"
     },
     "require-dev": {
         "fzaninotto/faker": "^1.8",
-        "phpstan/phpstan": "^0.11",
-        "orchestra/testbench": "^3.0",
-        "phpunit/phpunit": "^7.4",
-        "illuminate/support": "5.6.*|5.7.*|5.8.*",
+        "phpstan/phpstan": "^0.12",
```
Comment (matchish) [inline composer.json]: We don't need it. Larastan has it as a dependency.

File: composer.json [hunk 3]
```diff
@@ -17,17 +17,17 @@
...
+        "orchestra/testbench": "^4.0|^5.0",
+        "phpunit/phpunit": "^8.0|^9.0",
+        "illuminate/support": "^6.0|^7.0",
         "nunomaduro/larastan": "~0.3",
         "nette/bootstrap": "^2.4.6"
```
Comment (matchish) [inline composer.json]: Remove it plz. It was a temporal solution.

File: composer.json [hunk 4]
```diff
@@ -17,17 +17,17 @@
...
         "fzaninotto/faker": "^1.8",
-        "phpstan/phpstan": "^0.11",
-        "orchestra/testbench": "^3.0",
-        "phpunit/phpunit": "^7.4",
-        "illuminate/support": "5.6.*|5.7.*|5.8.*",
+        "phpstan/phpstan": "^0.12",
+        "orchestra/testbench": "^4.0|^5.0",
+        "phpunit/phpunit": "^8.0|^9.0",
+        "illuminate/support": "^6.0|^7.0",
         "nunomaduro/larastan": "~0.3",
```
Comment (matchish) [inline composer.json]: Let's update it to `~0.5`

**Comment by matchish** [comment]
There are phpstan errors still. Could you fix them? Please let me know if you have any trouble fixing any of them.

**Comment by burakcakirel** [comment]
I couldn't find the solution for the following error:
https://travis-ci.com/matchish/laravel-scout-elasticsearch/jobs/296449158#L700

File: src/ProgressReportable.php [hunk 5]
```diff
@@ -19,6 +19,6 @@ public function withProgressReport(ProgressBar $progressBar): void
 
     private function progressBar(): ProgressBar
     {
-        return $this->progressBar ?: new ProgressBar(new NullOutput());
+        return $this->progressBar;
```
Comment (matchish) [inline src/ProgressReportable.php]: This fix breaks some tests

File: src/Searchable/DefaultImportSource.php [hunk 6]
```diff
@@ -90,9 +90,6 @@ private function newQuery(): Builder
         }, $query);
     }
 
-    /**
-     * @return Builder[]|EloquentCollection
-     */
```
Comment (matchish) [inline src/Searchable/DefaultImportSource.php]: What is a reason to remove this doc comment?

File: src/ProgressReportable.php (hunk 5 above)
Comment (matchish) [reply src/ProgressReportable.php]: This line shouldn't be removed. I think the type of property should be specified as `ProgressBar|null`

File: src/Searchable/DefaultImportSource.php (hunk 6 above)
Comment (burakcakirel) [reply src/Searchable/DefaultImportSource.php]: It's already defined in its interface

**Comment by matchish** [comment]
The last phpstan error
```
  Line   Jobs/Stages/PullFromSource.php                                         
 ------ ----------------------------------------------------------------------- 
  29     Cannot access property $filter on                                      
         array<Illuminate\Database\Eloquent\Builder>|Illuminate\Database\Eloqu  
         ent\Collection.   
```

I think we should to add return type `Illuminate\Database\Eloquent\Collection` here 
https://github.com/matchish/laravel-scout-elasticsearch/blob/3d835ff941dc4dc274f9262dfae1a17e28a14bb5/src/Searchable/ImportSource.php#L15

**Comment by matchish** [comment]
Could you update PHP to `7.3` here?
https://github.com/matchish/laravel-scout-elasticsearch/blob/3d835ff941dc4dc274f9262dfae1a17e28a14bb5/.scrutinizer.yml#L13

**Comment by burakcakirel** [comment]
> The last phpstan error
> 
> ```
>   Line   Jobs/Stages/PullFromSource.php                                         
>  ------ ----------------------------------------------------------------------- 
>   29     Cannot access property $filter on                                      
>          array<Illuminate\Database\Eloquent\Builder>|Illuminate\Database\Eloqu  
>          ent\Collection.   
> ```
> 
> I think we should to add return type `Illuminate\Database\Eloquent\Collection` here
> https://github.com/matchish/laravel-scout-elasticsearch/blob/3d835ff941dc4dc274f9262dfae1a17e28a14bb5/src/Searchable/ImportSource.php#L15

Yes, you're right but If I add it, the new error occurs:
```
Line   Searchable/DefaultImportSource.php                                                                                              
 ------ -------------------------------------------------------------------------------------------------------------------------------- 
  94     Method Matchish\ScoutElasticSearch\Searchable\DefaultImportSource::get() should return Illuminate\Database\Eloquent\Collection  
         but returns array<Illuminate\Database\Eloquent\Builder>|Illuminate\Database\Eloquent\Collection.                                
 ------ --------------------------------------------------------------------------------------------------------------------------------
```

I think we should ignore it?

**Comment by matchish** [comment]
I think we can fix it 
`Illuminate\Database\Eloquent\Builder` return type here
https://github.com/matchish/laravel-scout-elasticsearch/blob/3d835ff941dc4dc274f9262dfae1a17e28a14bb5/src/Searchable/DefaultImportSource.php#L66
```
/** @var Illuminate\Database\Eloquent\Collection $models */
$models = $this->newQuery()->get();
return $models;
```

**Comment by matchish** [comment]
Cool! Great contribution! Could you bump version to `4.0.0` in `composer.json`, update `README.md` and `CHANGELOG.md`?

**Comment by burakcakirel** [comment]
> Don't forget to update requirements in README.md plz

It's already updated?:
https://github.com/matchish/laravel-scout-elasticsearch/pull/93/files#diff-04c6e90faac2675aa89e2176d2eec7d8R41-R42

**Comment by matchish** [comment]
You right. Sorry

File: phpstan.neon.dist [hunk 7]
```diff
@@ -5,6 +5,7 @@ parameters:
     paths:
         - src
     reportUnmatchedIgnoredErrors: false
+    inferPrivatePropertyTypeFromConstructor: true
```
Comment (matchish) [inline phpstan.neon.dist]: This one is unnecessary

**Comment by matchish** [comment]
There is two almost finished PR I want to merge them first before we bump to the new major version.

**Comment by Krystus** [comment]
@matchish keep up good work! Waiting for new version. 👍

**Pull request merged**